src/biogeophys/SoilStateInitTimeConstMod.F90

```diff
@@ -358,48 +364,22 @@ subroutine SoilStateInitTimeConst(bounds, soilstate_inst, nlfilename)
              soilstate_inst%csol_col(c,lev)= spval
           end do
 
-       else if (lun%urbpoi(l) .and. (col%itype(c) /= icol_road_perv) .and. (col%itype(c) /= icol_road_imperv) )then
-
-          ! Urban Roof, sunwall, shadewall properties set to special value
-          do lev = 1,nlevgrnd
-             soilstate_inst%watsat_col(c,lev) = spval
-             soilstate_inst%watfc_col(c,lev)  = spval
-             soilstate_inst%bsw_col(c,lev)    = spval
-             soilstate_inst%hksat_col(c,lev)  = spval
-             soilstate_inst%sucsat_col(c,lev) = spval
-             soilstate_inst%watdry_col(c,lev) = spval 
-             soilstate_inst%watopt_col(c,lev) = spval 
-             soilstate_inst%bd_col(c,lev) = spval 
-             if (lev <= nlevsoi) then
-                soilstate_inst%cellsand_col(c,lev) = spval
-                soilstate_inst%cellclay_col(c,lev) = spval
-                soilstate_inst%cellorg_col(c,lev)  = spval
-             end if
-          end do
-
-          do lev = 1,nlevgrnd
-             soilstate_inst%tkmg_col(c,lev)   = spval
-             soilstate_inst%tksatu_col(c,lev) = spval
-             soilstate_inst%tkdry_col(c,lev)  = spval
-             soilstate_inst%csol_col(c,lev)   = spval
-          end do
-
        else
 
           do lev = 1,nlevgrnd
              ! DML - this if statement could probably be removed and just the
              ! top part used for all soil layer structures
-             if ( soil_layerstruct /= '10SL_3.5m' )then ! apply soil texture from 10 layer input dataset 
+             if (.not. organic_frac_squared) then
```

Comment:
Your rework of `organic_fraq_squared` doesn't look quite right to me: `organic_frac_squared` .true. now seems to assume that we're using the same soil layer structure as on the file. Also, did you confirm that `organic_frac_squared` .false. gives the same answers as master when using 10SL_3.5m, since now you'd be using the interpolation code for that case (and my sense from the earlier discussion was that this combo isn't tested in our test suite)?